packages/agent-utils/src/llm/Chat.ts

```diff
@@ -1,95 +1,62 @@
-import { LlmApi, Tokenizer } from ".";
-import { Logger } from "../sys";
-
 import {
-  ChatCompletionRequestMessageRoleEnum,
-  ChatCompletionRequestMessage as ChatMessage
-} from "openai";
-
-export { ChatMessage };
-
-export type MessageType =
-  | "persistent"
-  | "temporary";
+  Tokenizer,
+  ChatLogs,
+  ChatMessage,
+  ChatLogType,
+  ContextWindow
+} from ".";
+import { Logger } from "../sys";
 
-interface MessageLog {
-  tokens: number;
-  msgs: ChatMessage[];
-}
+import { ChatCompletionRequestMessageRoleEnum } from "openai";
 
 export type ChatRole = ChatCompletionRequestMessageRoleEnum;
 
-export type ChatMessageLog = Record<MessageType, MessageLog>;
-
 export class Chat {
-  private _msgLogs: ChatMessageLog = {
-    "persistent": {
-      tokens: 0,
-      msgs: []
-    },
-    "temporary": {
-      tokens: 0,
-      msgs: []
-    }
-  };
-  private _maxContextTokens: number;
-  private _summaryTokens: number;
-  private _chunkTokens: number;
+  private _chatLogs: ChatLogs;
 
   constructor(
-    private _llm: LlmApi,
     private _tokenizer: Tokenizer,
+    private _contextWindow?: ContextWindow,
     private _logger?: Logger,
-  ) {
-    this._maxContextTokens = this._llm.getMaxContextTokens();
+  ) { }
 
-    // Summary size should be ~10% of total tokens
-    const summaryPerc = 0.10;
-    this._summaryTokens = Math.floor(
-      this._maxContextTokens * summaryPerc
-    );
-
-    // Chunk size should be ~70% of total tokens
-    const chunkPerc = 0.7;
-    this._chunkTokens = Math.floor(
-      this._maxContextTokens * chunkPerc
-    );
+  get chatLogs(): ChatLogs {
+    return this._chatLogs;
   }
 
   get tokens(): number {
-    return this._msgLogs["persistent"].tokens +
-      this._msgLogs["temporary"].tokens;
+    return this._chatLogs.tokens;
   }
 
   get tokenizer(): Tokenizer {
     return this._tokenizer;
   }
 
   get messages(): ChatMessage[] {
-    return [
-      ...this._msgLogs["persistent"].msgs,
-      ...this._msgLogs["temporary"].msgs
-    ];
+    return this._chatLogs.messages;
   }
 
   public add(
-    type: MessageType,
+    type: ChatLogType,
     msg: ChatMessage | ChatMessage[]
   ) {
-    const msgLog = this._msgLogs[type];
     let msgs = Array.isArray(msg) ? msg : [msg];
 
     for (const msg of msgs) {
       const tokens = this._tokenizer.encode(msg.content || "").length;
 
       // If the message is larger than the context window
```

Comment:
No it's still accurate. If a context window is present, we want to give it a chance to "chunk" a message that is above it's desired size.

Reply:
I'm being a nit, so feel free to ignore, but the message refers to "if the message is larger..." and the functionality `(this._contextWindow?.shouldChunk(tokens)) {` has no concept of "larger" (it's an internal implementation detail)

Reply:
Ah okay, I see what you mean. So a better comment would be something like "See if the context window needs to chunk the message"

Reply:
Yup, exactly.
Or just removing the comment, since it's already evident from the code.
For the most part (99%) I really only believe in comments that explain "why" the code does what it does instead of "what" the code does (cause the latter should always be achievable through cleaner code)